main.py

```diff
@@ -0,0 +1,136 @@
+    # train
+    parser.add_argument('--seed', type=int, help='Random seed (default: %(default)s)')
+    parser.add_argument('--epochs', type=int, default=10000, help='Number of epochs to train (default: %(default)s)')
+    parser.add_argument('--batch_size', type=int, default=16, help='Size of training batches (default: %(default)s)')
+    parser.add_argument('--optimizer', default='adam', choices=['adam', 'sgd'], help='Optimizer: SGD or Adam (default: %(default)s)')
+    parser.add_argument('--learning_rate', type=float, default=0.0001, help='Learning rate for optimizer (default: %(default)s)')
+    parser.add_argument('--weight_decay', type=float, default=0, help='Weight decay factor (default: %(default)s)')
+    parser.add_argument('--momentum', type=float, default=0.9, help='Momentum factor for SGD only (default: %(default)s)')
+    parser.add_argument('--patience', type=int, default=5, help='Number of epochs to wait for improvement before early stopping (default: %(default)s)')
+
+    # model
+    parser.add_argument('--model_name', default='cnn', choices=['caml', 'cnn'], help='Model to be used (default: %(default)s)')
+    # default
+    parser.add_argument('--num_filter_maps', type=int, default=128, help='Number of filters in convolutional layers (default: %(default)s)')
+    parser.add_argument('--filter_size', type=int, default=4, help='Size of convolutional filter (default: %(default)s)')
+    parser.add_argument('--dropout', type=float, default=0.2, help='Optional specification of dropout (default: %(default)s)')
+
+    # eval
+    parser.add_argument('--eval_batch_size', type=int, default=256, help='Size of evaluating batches (default: %(default)s)')
+    parser.add_argument('--metrics_thresholds', type=float, nargs='+', default=[0.5], help='Thresholds to monitor for metrics (default: %(default)s)')
+    parser.add_argument('--monitor_metrics', nargs='+', default=['P@1', 'P@3', 'P@5'], help='Metrics to monitor while validating (default: %(default)s)')
+    parser.add_argument('--val_metric', default='P@1', help='The metric to monitor for early stopping (default: %(default)s)')
+
+    # pretrained vocab / embeddings
+    parser.add_argument('--vocab_file', type=str, help='Path to a file holding vocabuaries (default: %(default)s)')
+    parser.add_argument('--embed_file', type=str, help='Path to a file holding pre-trained embeddings (default: %(default)s)')
+
+    # others
+    parser.add_argument('--cpu', action='store_true', help='Disable CUDA')
+    parser.add_argument('--display_iter', type=int, default=100, help='Log state after every n epochs (default: %(default)s)')
+    parser.add_argument('--data_workers', type=int, default=1, help='Use multi-cpu core for data pre-processing (default: %(default)s)')
+    parser.add_argument('--eval', action='store_true', help='Only run evaluation on the test set (default: %(default)s)')
+    parser.add_argument('--load_checkpoint', help='The checkpoint to warm-up with (default: %(default)s)')
```

Comment:
``load_checkpoint`` is more like an action to me. Rename to, for example `checkpoint_path`, to match the description `The checkpoint to warm-up with`?